### PR TITLE
Add new way of authentication

### PR DESCRIPTION
**Pull request by tehAgitto**
# Example usage: 
```python
base_url = "https://sandbox-sdwan-2.cisco.com/"
username = "devnetuser"
password = "RG!_Yw919_83"

s = requests.Session()
s.auth = vManageAuth(base_url, username, password, verify=False)
s.verify = False

response = s.get("https://sandbox-sdwan-2.cisco.com/dataservice/device")
print(response.json()['data'])
>>> [{'deviceId': '10.10.1.1', 'system-ip': '10.10.1.1', 'host-name': 'vmanage', 'reachability': 'reachable', ...}]
```

### CODE REVIEW

**Review by JimOverholt — DISMISSED** [review]

File: vmngclient/vmanage_auth.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    """Exception raised for invalid credentials
+
+    Attributes:
+        username (str): vManage username.
+        password (str): vManage password.
+        message (str): precise error explanation.
+    """
+
+    def __init__(
+        self, username: str, password: str, message: str = "Username and/or password is incorrect. Please try again!"
```
Comment (tomaszwarda) [inline vmngclient/vmanage_auth.py:20]: The message from vmanage is:
image
Shouldn't we include same options of invalid login?

File: vmngclient/vmanage_auth.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    """Attaches vManage Authentication to the given Requests object.
+
+    vManage REST API access control is based on sessions.
+    The call method do whatever is required to make the vManage authentication work.
+    The following are typical steps for a user to consume the API:
+    1. Log in with a user name and password to establish a session.
+    2. Get a cross-site request forgery prevention token, which is required for most POST operations.
+
+    Attributes:
+        base_url (str): url with port f.e. https://1.1.1.1:1111
```
Comment (tomaszwarda) [inline vmngclient/vmanage_auth.py:42]:
```suggestion
        base_url (str): url (with port if applicable) f.e. https://1.1.1.1:1111
```

**Review by tomaszwarda — APPROVED** [review]

File: vmngclient/vmanage_auth.py (hunk 1 above)
Comment (JimOverholt) [reply vmngclient/vmanage_auth.py:20]: Yes, it wold be nice to have a negative test available.

Comment (tehAgitto) [reply vmngclient/vmanage_auth.py:20]: Moved to #45 Thanks!

**Review by sbasan — APPROVED** [review]
lgtm

**Pull request merged**